src/targets/gpu/hip.cpp

```diff
@@ -246,6 +247,14 @@ void gpu_sync()
 
 void gpu_sync(const context& ctx) { ctx.finish(); }
 
+void hip_async_memset(context& ctx, const argument& dst, int value)
+{
+    std::size_t dst_size = dst.get_shape().bytes();
+    auto status          = hipMemsetAsync(dst.data(), value, dst_size, ctx.get_stream().get());
```

Comment:
Do we not need to add any sort of event syncronization or are we assuming things are handled by using the context's stream, other items will sync to the end of this MemSet?

Reply:
The memset should be enqued on the stream so we dont need any extra synchronization(the queue for the stream is always in order).